src/pages/items/components/allitem.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useEffect } from "react";
+import "./allitem.css";
+import heart from "../../../assets/item/ic_heart.svg";
+import search from "../../../assets/item/ic_search.svg";
+import dropdown from "../../../assets/item/ic_arrow_down.svg";
+import { fetchProducts } from "../../../apis/api.tsx";
+
+interface Product {
+  id: number;
+  name: string;
+  price: number;
+  description: string;
+  favoriteCount: number;
+  images: string[];
+  createdAt: string;
+}
+
+const ITEMS_PER_PAGE = 10;
+
+export default function AllItem() {
+  const [showDropdown, setShowDropdown] = useState<boolean>(false);
+  const [sortBy, setSortBy] = useState<string>("최신순");
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [sortedProducts, setSortedProducts] = useState<Product[]>([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const data = await fetchProducts(sortBy);
+        setSortedProducts(data.list);
+      } catch (error) {
+        console.error("Error fetching data:", error);
+      }
+    };
+
+    fetchData();
+  }, [sortBy, currentPage]);
+
+  const toggleDropdown = () => {
+    setShowDropdown(!showDropdown);
+  };
+
+  const handleSortChange = (sortOption: string) => {
+    setSortBy(sortOption);
+    console.log("sortBy 상태:", sortOption);
+    toggleDropdown();
+  };
+
+  const sortOptions = ["최신순", "좋아요순"];
+
+  const getCurrentItems = () => {
+    const startIndex = (currentPage - 1) * ITEMS_PER_PAGE;
+    const endIndex = startIndex + ITEMS_PER_PAGE;
+    return sortedProducts.slice(startIndex, endIndex);
+  };
```

Comment:
이 함수가 어떻게 보면 state를 활용해 목록화를 하기위한 계산을 담당하고 있는거네요? 

쓰이는 쪽의 코드를 보면  getCurrentItems().map 과 같이 함수를 실행하고 바로 리턴값을 받아 map을 실행하고있는데, 10개의 array item을 조정하는 연산이다보니 상대적으로 목록이 크다고 볼수있고, 계산도 복잡한편이네요 :) 

useMemo hook으로 리턴을 받아 컴포넌트 body 내부에서는 해당 결과값을 사용해주는 형태는 어떨까요? 
부수적인 효과로 가독성도 괜찮아질것같아요! :) 
```suggestion
  const currentItems = useMemo(() => { 
   // 대소문자 구별도 덤으로 없애줍시다!
    const filteredProducts = sortedProducts.filter(product =>
      product.name.toLowerCase().includes(searchTerm.toLowerCase())
    );

    const startIndex = (currentPage - 1) * ITEMS_PER_PAGE;
    const endIndex = startIndex + ITEMS_PER_PAGE;
    return filteredProducts.slice(startIndex, endIndex);
  }, [sortedProducts, currentPage, searchTerm]);
```